bld/CLMBuildNamelist.pm

```diff
@@ -901,6 +901,12 @@ sub setup_cmdl_bgc {
            $log->warning("$var normally use_nitrif_denitrif should only be FALSE if FATES is on, it has NOT been validated for being off for BGC mode" );
         }
      }
+     # if MIMICS is on and use_fates = .true. then use_lch4 must = .true.
+     if ( (! &value_is_true($nl_flags->{'use_lch4'})) && &value_is_true($nl_flags->{'use_fates'}) ) {
+        if ( $soil_decomp_method eq "MIMICSWieder2015" ) {
+           $log->warning("If MIMICS is on and use_fates = .true. then use_lch4 must be .true. and currently it's not" );
```

Comment:
@slevisconsulting @ekluzek will this trigger a warning with or without a graceful fail?  Seems we need the former right?

Reply:
Yes, I agree and that's what we have. The warning will abort by default with the above message. You can get around it if you tell build-namelist to ignore-warnings. It'll then still print the warning, but will build the namelist and you can run the model. I'm not sure what the code will do though if you run with FATES and MIMICS without methane? Will it just give bad results or will it abort in some strange way? Do we also need an abort in the FORTRAN code as well?

Reply:
I see what you're saying @ekluzek ...if use_lch4 = .false.:
- This will abort with the error gracefully by default, which I have confirmed, BUT
- if user bypasses the error with `ignore-warnings`, the message will only appear as a warning. I don't know off the top of my head whether the run will abort, but it will certainly give bad results because it won't have access to the annsum_npp variable from FATES.
Let me know how you prefer to resolve this. Do we have error checks in the fortran for all error checks that we bypass with `ignore-warnings`? Or do we assume that users take matters in their own hands when they `ignore-warnings`?

Reply:
We sometimes both have an exit in the build-namelist as well as the FORTRAN code. And sometimes we just have it in the build-namelist, because it's easier. There are also things that the build-namelist can't really catch, so it's only in the FORTRAN code.

But, yes I think it's reasonable to have the stance that if the user sets "ignore-warnings" that they are on their own and get what they deserve. I do think that the worst kind of problem to have though is silently giving bad results. That's expensive in terms of human and computer time. So I can go either way on this in terms of adding a check in the FORTRAN code.

Reply:
I decided to add an error check in the fortran.